core/base/common/BaseClass.h

```diff
@@ -62,6 +62,11 @@ namespace ttk {
 
   class BaseClass {
   public:
+#if TTK_ENABLE_MPI
+    long int *GlobalIdsArray;
+    unsigned char *PointGhostArray;
```

Comment:
Based on the discussion with @m-s-will , I'd recommend to remove that array and only use the `rankArray_` instead.